router/router.go

```diff
@@ -0,0 +1,188 @@
+package router
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/go-etcd/etcd"
+
+	"github.com/spesnova/iruka/schema"
+)
+
+const (
+	DefaultKeyPrefix = "/vulcand"
+	DefaultMachines  = "http://localhost:4001"
+)
+
+type Router struct {
+	etcd      etcd.Client
+	keyPrefix string
+}
+
+func NewRouter(machines, keyPrefix string) *Router {
+	if os.Getenv("IRUKA_ETCD_MACHINES") != "" {
+		machines = os.Getenv("IRUKA_ETCD_MACHINES")
+	}
+
+	if machines == "" {
+		machines = DefaultMachines
+	}
+
+	m := strings.Split(machines, ",")
+	etcdClient := *etcd.NewClient(m)
+	return &Router{etcdClient, keyPrefix}
+}
+
+// TODO(dtan4):
+//   Divide by upstream (blue/green)
+func (r *Router) AddBackend(name string) error {
```

Comment:
I think `backendID` is more understandable than `name`.
I couldn't understand what `name` means at first time.